hpx/runtime/components/unwrapping_result_policy.hpp

```diff
@@ -29,35 +29,17 @@ namespace hpx { namespace components
     /// This class is a distribution policy that can be using with actions that
     /// return futures. For those actions it is possible to apply certain
     /// optimizations if the action is invoked synchronously.
-    struct unwrapping_result_policy : target_distribution_policy
+    struct unwrapping_result_policy
     {
     public:
-        /// Default-construct a new instance of a \a unwrapping_result_policy.
-        /// This policy will represent the local locality.
-        unwrapping_result_policy() = default;
-
-        /// Create a new \a unwrapping_result_policy representing the
-        /// target locality
-        ///
-        /// \param id     [in] The global address of the target object
-        ///
-        unwrapping_result_policy operator()(id_type const& id) const
-        {
-            return unwrapping_result_policy(id);
-        }
+        unwrapping_result_policy(id_type const& id)
```

Comment:
I don't particularly like this change. This leads to a copy of the id_type. Wouldn't it we better if we move it into the async call instead? What about lifetime?

Reply:
No, unwrapping_policy holds the id by reference. No copy is happening anymore (at least that is the intent...).

Reply:
The copy is happening in the call to async (and friends).

Reply:
Huh? Could you elaborate?

Reply:
These calls: https://github.com/STEllAR-GROUP/hpx/pull/3465/files#diff-a1d7921ce43cab0e4a7fb54f51558858R85

Do capture the id, don't they? I might be totally off here, but it looks like it could be moved there as well.

Reply:
@sithhell so do I understand correctly that you propose to convert the `unwrapping_result_policy` into a one-shot facility, allowing to move the `id_type` to be possibly moved through into the function object created by async? If so, that would require that we can move the initial `id_type` instance as well  (the one the `unwrapping_result_policy` is created from), which is not the case most of the time, I belive.

Reply:
After this discussing this offline, @sithhell agreed for this to go forward as is.